src/ophyd_async/core/signal.py

```diff
@@ -63,18 +63,38 @@ def __init__(
     ) -> None:
         self._timeout = timeout
         self._initial_backend = self._backend = backend
+
         super().__init__(name)
 
     async def connect(
         self, mock=False, timeout=DEFAULT_TIMEOUT, force_reconnect: bool = False
     ):
+        """Connect self with a timeout
+
+        Args:
+            mock (bool, optional): _description_. Defaults to False.
+            timeout (_type_, optional): _description_. Defaults to DEFAULT_TIMEOUT.
+            force_reconnect (bool, optional): _description_. Defaults to False.
+        """
+        can_use_previous_connection: bool = (
+            self._connect_task and not (self._connect_task.done() and self._connect_task.exception())
+        )
```

Comment:
```suggestion
        can_use_previous_connect = (
            self._connect_task
            and not (self._connect_task.done() and self._connect_task.exception())
            and self._connect_mock_arg == mock
        )
```

Reply:
not sure how would the mock work here. the commits pushed were just the draft - tbh I should have marked the PR as draft

Reply:
We can use the previous connect if it is in progress or successful, and it used the same mock arg

Reply:
that I understand, what about mocking?